app/pages/profile/comment-link.cjsx

```diff
@@ -57,13 +57,13 @@ module.exports = React.createClass
           <span>
             {' '}in{' '}
             <a href={@state.href}>
-              {if @state.project? and @state.owner
+              {if @state.project? and @state.owner? and !@props.project?
```

Comment:
@state.project is the project to which the comment belongs.
@props.project is the project whose context you are currently viewing things in. 

They are not necessarily the same.